app/assets/javascripts/reducers/articles.js

```diff
@@ -61,6 +67,15 @@ export default function articles(state = initialState, action) {
       }
       return { ...state, wikiFilter: action.wiki };
     }
+    case SET_NEWNESS_FILTER: {
+      switch (action.newness) {
+        case 'new':
+        case 'existing':
+          return { ...state, newnessFilter: action.newness };
+        default:
+          return state;
```

Comment:
Here's the bug. Changing from one of the other filters to 'both' falls through to the default case and does not change state.

I suggest matching the implementation for SET_PROJECT_FILTER:

```javascript
    case SET_NEWNESS_FILTER: {
      if (action.newness === 'both') {
        return { ...state, newnessFilter: null };
      }
      return { ...state, newnessFilter: action.newness };
    }
```

In this case, because of how you implemented the selector, it was also work to just do it unconditionally:

```javascript
    case SET_NEWNESS_FILTER: {
      return { ...state, newnessFilter: action.newness };
    }
```

That could even be extended to the `SET_PROJECT_FILTER` case, if we tweak the selector implementation to use a case statement with the default fallback like in new selector.

Reply:
Yes, I've edited it so that we could use a simple reducer.